src/peer_connection.cpp

```diff
@@ -168,7 +168,8 @@ namespace libtorrent {
 		, m_exceeded_limit(false)
 		, m_slow_start(true)
 	{
-		m_counters.inc_stats_counter(counters::num_tcp_peers + socket_type_idx(*m_socket) - 1);
+		m_counters.inc_stats_counter(counters::num_tcp_peers
+			+ static_cast<std::uint8_t>(socket_type_idx(*m_socket)));
```

Comment:
removing the `-1` here is subtle. In the new enum, there is no value representing an "empty" socket, so the numbers shifted down one step